src/libraries/System.Security.Cryptography/tests/ShakeTestDriver.cs

```diff
@@ -580,5 +797,18 @@ static void ThreadWork(object obj)
                 }
             }
         }
+
+        private static void ReadChunked(TShake shake, Span<byte> destination)
+        {
+            int read = 0;
+            int outputLength = destination.Length;
+
+            while (read < outputLength)
+            {
+                int size = Math.Min(Math.Max(outputLength / 4, 1), outputLength - read);
```

Comment:
byte-at-a-time was too slow for all of the test vectors, so do a quarter read if the input is >= 4.